banking_system/command_line_interface.cpp

```diff
@@ -0,0 +1,127 @@
+#include "command_line_interface.h"
+
+#include <iostream>
+#include <string>
+#include <memory>
+#include <utility>
+
+#include "funds_eur.h"
+#include "funds_pln.h"
+#include "funds_usd.h"
+#include "maintenance.h"
+
+namespace bank_cli {
+    void CommandLineInterface::createNewClientProfile(bank::Maintenance& maintenance) {
+        std::string firstName, middleName, lastName;
+        std::cout << "Provide a first name:" << std::endl;
+        std::cin >> firstName;
+        std::cout << "Provide a middle name:" << std::endl;
+        std::cin >> middleName;
+        std::cout << "Provide a last name:" << std::endl;
+        std::cin >> lastName;
+        maintenance.addClient(std::make_shared<bank::Person>(firstName, middleName, lastName));
+    }
+
+    void CommandLineInterface::depositMoney(std::unique_ptr<bank::IManager> const& accountManager, std::shared_ptr<bank::IAccount> const& account) {
```

Comment:
Let's avoid here to use reference to unique_ptr and use const reference instead, or return value from method maybe?